### DISCUSSION TITLE
Prevent AdGuard from sleeping

### DISCUSSION THREAD

**Discussion by jxwinks** [AdguardTeam/AdGuardHome#3770] [Q&A]
I've set my Primary DNS to be my home iMac in order to use AdGuard Home across ALL my devices.

What should my iMac computer settings (preferences) be if I want to make sure AdGuard doesn't sleep and prevent other devices from accessing the Internet?

**Comment by ainar-g** [comment]
Your DNS server should generally be a machine that is as close to being always-online as possible.  A typical home computer that sleeps or hibernates is therefore just not a good choice, generally speaking.

**Reply by jxwinks** [reply]
I understand that. Do you have any suggestions on iMac settings? Maybe some specific Energy Saver preferences?

Is there some other type of machine on which AdGuard can be installed that would be a better choice?

**Reply by ainar-g** [reply]
I personally don't use Apple products, sorry.  But I'm pretty sure somebody else could chime in about that.

A lot of people use relatively low-power machines, like a Raspberry Pi.  If you want an Apple product, a Mac mini could also probably work well for this use case.
